### PR TITLE
Category axis labels for chart.js

### PR DESCRIPTION
**Pull request by leapwill**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
   - In the documentation [here](https://www.chartjs.org/docs/latest/axes/cartesian/category.html#category-axis-definition), under "As part of axis definition".
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
   - Not a new version
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

As a side note, Prettier changed lots of things in the chart.js files and I decided to not include those hunks in this PR.

### CODE REVIEW

File: types/chart.js/index.d.ts [hunk 1]
```diff
@@ -729,6 +729,7 @@ declare namespace Chart {
         type?: ScaleType | string;
         display?: boolean | string;
         id?: string;
+        labels?: Array<string | string[] | number | number[] | Date | Date[] | Moment | Moment[]>;
```
Comment (lekoaf) [inline types/chart.js/index.d.ts:732]: According to the documentation this property can be `string[]`, so why all the other types? Especially the nested arrays.

Comment (leapwill) [reply types/chart.js/index.d.ts:732]: Same as `ChartData.labels`, and these types produce valid labels on the chart. Other issues ([one here](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/38583#issuecomment-535384241)) mention "The chart.js documentation is quite a mess", which I don't fully agree with, but I don't think it accurately describes all behavior of the library.

Comment (IVIosi) [reply types/chart.js/index.d.ts:732]: @leapwill As discussed [here](https://github.com/chartjs/Chart.js/issues/7933#issuecomment-714309035), labels' types for scales are different from category axis labels. So `string[]` is the correct type for labels.

**Review by IVIosi — APPROVED** [review]

**Comment by leapwill** [comment]
Ready to merge

**Review by IVIosi — APPROVED** [review]

**Comment by IVIosi** [comment]
Ready to merge

**Pull request merged**